src/Compiler/Compiler.cpp

```diff
@@ -1499,6 +1494,15 @@ std::string Compiler::ReplaceAll(std::string_view str, std::string_view from,
   return result;
 }
 
+std::pair<std::filesystem::path, int> Compiler::CurrentPath(
```

Comment:
Why do you need to ask Tcl what is the current path?
There is a simple std::filesystem::path API to do so.

Reply:
All from C++, no need to do a roundtrip to Tcl.

Reply:
auto path = std::filesystem::current_path();  
If you need to save the current path in the compiler class, you should do that when initializing the class.

Reply:
Right! I thought it will not work when do **cd** somewhere in the tcl console but for some reason it works.